File: backend/src/main/java/com/project/entities/Appointment.java
package com.project.entities;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Entity
@Table(name = "appointments",
       indexes = {
           @Index(columnList = "professional_id,startTime")
       })
public class Appointment extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long appointmentId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "professional_id", nullable = false)
    private Professional professional;

    @NotNull
    private LocalDateTime startTime;

    @NotNull
    private LocalDateTime endTime;

    @Enumerated(EnumType.STRING)
    private AppointmentStatus status = AppointmentStatus.PENDING;

    @Size(max = 255)
    private String cancellationReason;
}

File: backend_dotnet/SafeMind.Core/Entities/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using SafeMind.Core.Enums;

namespace SafeMind.Core.Entities
{
    [Table("appointments")]
    public class Appointment : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("appointment_id")]
        [JsonPropertyName("appointmentId")]
        public long AppointmentId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        [Column("user_id")]
        [JsonPropertyName("userId")]
        public long UserId { get; set; }

        [JsonPropertyName("user")]
        public User User { get; set; } = null!;

        [Required]
        [ForeignKey(nameof(Professional))]
        [Column("professional_id")]
        [JsonPropertyName("professionalId")]
        public long ProfessionalId { get; set; }

        [JsonPropertyName("professional")]
        public Professional Professional { get; set; } = null!;

        [Required]
        [Column("start_time")]
        [JsonPropertyName("startTime")]
        public DateTime StartTime { get; set; }

        [Required]
        [Column("end_time")]
        [JsonPropertyName("endTime")]
        public DateTime EndTime { get; set; }

        [Required]
        [Column("status")]
        [JsonPropertyName("status")]
        public AppointmentStatus Status { get; set; } = AppointmentStatus.PENDING;

        [MaxLength(255)]
        [Column("cancellation_reason")]
        [JsonPropertyName("cancellationReason")]
        public string? CancellationReason { get; set; }
    }
}
